Applets/UI/Onboarding/OnboardingViews/OnBoarding.swift

```diff
@@ -0,0 +1,172 @@
+                    }
+
+                }
+
+                HStack {
+                    Button {
+                        firstRun = false
+                    } label: {
+                        Text("Skip")
+                            .fontWeight(.semibold)
+                            .foregroundColor(.white)
+                    }
+                    
+                    // Indicators
+                    HStack(spacing: 8) {
+                        ForEach(BoardingScreen.boardingScreens.indices, id: \.self) { index in
+                            Circle()
+                                .fill(.white)
+                                .opacity(index == getIndex() ? 1 : 0.4)
+                                .frame(width: 8, height: 8)
+                                .scaleEffect(index == (getIndex()) ? 1.3 : 0.85)
+                                .animation(.easeInOut, value: getIndex())
+                        }
+                    }
+                    .frame(maxWidth: .infinity)
+
+                    Group {
+                        if offset < min(offset + getScreenBounds().width, getScreenBounds().width * 3) {
+
+                            // This way of animating the screens needs to change, maybe do an enum for each screen. The following button will cause the Finish button to flash up for an instant before the animation, and the only way I can think of to fix it is to refactor the whole thing.
+                            Button {
+                                // Set Max Offset
+                                // Maximum of 4 screens, so max will be 3 * width
+                                offset = min(offset + getScreenBounds().width, getScreenBounds().width * 3)
+
+                            } label: {
+                                Text("Next")
+                            }
+
+                        } else {
+                            Button {
+                                firstRun = false
+                            } label: {
+                                Text("Finish")
+
+                            }
+                        }
+
+                    }
+                    .fontWeight(.semibold)
+                    .foregroundColor(.white)
+
+                }
+                .padding(.top, 30)
+                .padding(.horizontal, 8)
+            }
+            .padding()
+            , alignment: .bottom
+        )
+    }
+    
+    // Get rotation
+    func getRotation() -> Double {
+        let progress = offset / (getScreenBounds().width * 4)
+        let rotation = Double(progress) * 360         // Do one full rotation (can also do less)
+        
+        return rotation
+    }
+    
+    
+    // Change the background colour based on the offset
+    func getIndex() -> Int {
+        let progress = (offset / getScreenBounds().width).rounded()
+        
+        return Int(progress)
+    }
+}
+
+// Extension of View to get screen bounds
+extension View {
```

Comment:
And this extension also make separate because we can use sam in other views in order to make sizes fit to different screen sizes